nucliadb/nucliadb/ingest/consumer/shard_creator.py

```diff
@@ -89,9 +90,17 @@ async def handle_message(self, raw_data) -> None:
     @metrics.handler_histo.wrap({"type": "shard_creator"})
     async def process_kb(self, kbid: str) -> None:
         logger.info({"message": "Processing notification for kbid", "kbid": kbid})
-        kb_shards = await self.shard_manager.get_shards_by_kbid_inner(kbid)
-        current_shard: writer_pb2.ShardObject = kb_shards.shards[kb_shards.actual]
+        async with self.driver.transaction(read_only=True) as txn:
+            kb_shards = await shards_data_manager.get_kb_shards(txn, kbid)
+            current_shard = await self.shard_manager.get_current_active_shard(txn, kbid)
 
+        if kb_shards is None or current_shard is None:
+            msg = "Processing a notification for a nonexistent"
+            logger.error(msg, extra={"kbid": kbid})
+            raise Exception(msg)
+
+        # TODO: when multiple shards are allowed, this should either handle the
```

Comment:
we could start simple for now and keep the same strategy: when the shard that has been written to goes over the max paragraph/fields size, we set it as read_only and create another one.

Reply:
we should keep the other ones open though

Reply:
Why would you want to leave the old one open? The behavior right now in this PR should be the same as before (the TODO is just a pointer for the future). Currently, increasing the `actual` field when a new shard is added, the other is automaticall "closed"